src/main/java/ch/njol/skript/expressions/ExprInventory.java

```diff
@@ -39,25 +51,86 @@
 		"clear the player's inventory",
 		"remove 5 wool from the inventory of the clicked block"})
 @Since("1.0")
-public class ExprInventory extends SimplePropertyExpression<InventoryHolder, Inventory> {
+public class ExprInventory extends SimpleExpression<Object> {
+
+	private boolean inLoop;
+	@SuppressWarnings("null")
+	private Expression<InventoryHolder> holders;
+
 	static {
-		register(ExprInventory.class, Inventory.class, "inventor(y|ies)", "inventoryholders");
+		PropertyExpression.register(ExprInventory.class, Object.class, "inventor(y|ies)", "inventoryholders");
 	}
-	
+
 	@Override
-	@Nullable
-	public Inventory convert(final InventoryHolder h) {
-		return h.getInventory();
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		// if we're dealing with a loop of just this expression
+		Node n = SkriptLogger.getNode();
+		inLoop = n != null && ("loop " + parseResult.expr).equals(n.getKey());
```

Comment:
Do not check for node's key in expressions. It could *really* screw up things if logging system is changed. Which might happen, because current one has, hmm, some issues.

Proper solution would be making parser to expose the information in parse result. Or perhaps even a static field/method in ScriptLoader.

Reply:
how? All the method does is return the current node as set by SkriptLogger#setNode, and it should not do anything more than that. It's current functionality is incredibly useful for things like this and addons that wish to make things like skript's command system or otherwise use SectionValidator.